index.d.ts

```diff
@@ -93,8 +93,8 @@ export function combineReducers<S>(reducers: ReducersMapObject<S>): Reducer<S>;
  * transform, delay, ignore, or otherwise interpret actions or async actions
  * before passing them to the next middleware.
  */
-export interface Dispatch<S> {
-    <A extends Action>(action: A): A;
+export interface Dispatch<A extends Action = Action> {
```

Comment:
Yes, `redux-thunk` and `redux-promise` have different type for Dispatch. 
But I think they should be re-defined in that packages.
Or just use  `any` here :-)